flask_rebar/rebar.py

```diff
@@ -83,36 +81,30 @@ def wrapped(*args, **kwargs):
 
         rv = f(*args, **kwargs)
 
-        if marshal_schema:
-            if isinstance(rv, tuple):
-                data, status_code = rv[0], rv[1]
-            else:
-                data, status_code = rv, 200
-
-            try:
-                schema = marshal_schema[status_code]
-            except KeyError:
-                raise
-
-            if schema is None:
-                # Allow for the schema to be declared as None, which allows
-                # for status codes with no bodies (i.e. a 204 status code)
-                return response(
-                    data=data, status_code=status_code
-                )
+        if not marshal_schema:
+            return rv
 
-            marshaled = marshal(
-                data=data,
-                schema=schema
-            )
+        data, status_code, headers = rv, 200, None
+        if isinstance(rv, tuple):
```

Comment:
Just found Flask's own code for handling tuple responses:
https://github.com/pallets/flask/blob/7e714bd28b6e96d82b2848b48cf8ff48b517b09b/flask/app.py#L1932

Would probably be better to match that more closely here, or maybe better yet just call it and then modify the result as needed.